src/ui/SearchInterface/SearchInterface.ts

```diff
@@ -1230,10 +1214,38 @@ export class StandaloneSearchInterface extends SearchInterface {
     // By using a setTimeout, we allow other possible code related to the search box / magic box time to complete.
     // eg: onblur of the magic box.
     setTimeout(() => {
-      this._window.location.href = `${link.protocol}//${link.host}${pathname}${link.search}${hash}${HashUtils.encodeValues(stateValues)}`;
+      this._window.location.href = `${link.protocol}//${link.host}${pathname}${link.search}${hash}${this.encodedHashValues}`;
     }, 0);
   }
 
+  private get encodedHashValues() {
+    const values = this.modelAttributesToIncludeInUrl;
+    const uaCausedBy = this.uaCausedBy;
+    const uaMeta = this.usageAnalytics.getCurrentEventMeta();
+
+    if (uaCausedBy != null) {
+      values['firstQueryCause'] = uaCausedBy;
+    }
+
+    if (uaMeta != null && !isEmpty(uaMeta)) {
+      values['firstQueryMeta'] = uaMeta;
+    }
+
+    return HashUtils.encodeValues(values);
+  }
+
+  private get modelAttributesToIncludeInUrl() {
+    const usingLocalStorageHistory = this.historyManager instanceof LocalStorageHistoryController;
```

Comment:
Why not simply `this.options.useLocalStorageForHistory` ?

Reply:
For localstorage history to be active, two options need to be set to `true`: `enableHistory` and `useLocalStorageForHistory`.

I could have written a condition that checks both. However, I find this approach assumes a certain implementation in the `SearchInterface` class. Looking at the instance directly serves as a better source of truth of the state of the interface, I feel.